pkg/apis/internal.admin.acorn.io/v1/computeclassresources.go

```diff
@@ -0,0 +1,144 @@
+	for computeClass, resources := range incoming {
+		if _, ok := current[computeClass]; !ok {
+			current[computeClass] = ComputeResources{}
+		}
+
+		c := current[computeClass]
+		c.Memory = AddQuantity(c.Memory, resources.Memory)
+		c.CPU = AddQuantity(c.CPU, resources.CPU)
+		current[computeClass] = c
+	}
+}
+
+// Remove will remove the ComputeClassResources of another ComputeClassResources struct from the current one. Calling remove
+// will be a no-op for any resource values that are set to unlimited.
+func (current ComputeClassResources) Remove(incoming ComputeClassResources) {
+	for computeClass, resources := range incoming {
+		if _, ok := current[computeClass]; !ok || current == nil {
+			continue
+		}
+
+		c := current[computeClass]
+		c.Memory = SubQuantity(c.Memory, resources.Memory)
+		c.CPU = SubQuantity(c.CPU, resources.CPU)
+		current[computeClass] = c
+
+		// Don't keep empty ComputeClasses
+		if c.Equals(ComputeResources{}) {
+			delete(current, computeClass)
+		}
+	}
+}
+
+// Fits will check if a group of ComputeClassResources will be able to contain
+// another group of ComputeClassResources. If the ComputeClassResources are not able to fit,
+// an aggregated error will be returned with all exceeded ComputeClassResources.
+// If the current ComputeClassResources defines unlimited, then it will always fit.
+func (current ComputeClassResources) Fits(incoming ComputeClassResources) error {
+	var exceededResources []string
+
+	// Check if any of the quantity resources are exceeded
+	for computeClass, resources := range incoming {
+		// If a specific compute class is defined on current then we check if it will
+		// fit the incoming resources. If is not defined, then we check if the current
+		// resources has AllComputeClasses defined and if so, we check if the incoming
+		// resources will fit those.
+		if _, ok := current[computeClass]; !ok {
+			if _, ok := current[AllComputeClasses]; !ok {
+				continue
+			}
+			computeClass = AllComputeClasses
+		}
+
+		for _, r := range []struct {
+			resource          string
+			current, incoming resource.Quantity
+		}{
+			{"Memory", current[computeClass].Memory, resources.Memory},
+			{"CPU", current[computeClass].CPU, resources.CPU},
+		} {
+			if !FitsQuantity(r.current, r.incoming) {
+				exceededResources = append(exceededResources, fmt.Sprintf("%s:%s", computeClass, r.resource))
+			}
+		}
+	}
+
+	// Build an aggregated error message for the exceeded resources
+	if len(exceededResources) > 0 {
+		return fmt.Errorf("%w: %s", ErrExceededResources, strings.Join(exceededResources, ", "))
+	}
+
+	return nil
+}
+
+// ToString will return a string representation of the ComputeClassResources within the struct.
+func (current ComputeClassResources) ToString() string {
+	var resourceStrings []string
+
+	for computeClass, resources := range current {
+		resourceStrings = append(resourceStrings, fmt.Sprintf("Computeclass \"%s\": { %s }", computeClass, resources.ToString()))
```

Comment:
Nit: you can use `%q` to add double quote

Reply:
Oh yeah, good point! I forgot that exists.